graph.py

```diff
@@ -0,0 +1,125 @@
+class Graph(object):
+    """
+    Create the graph class and initialize an empty dictionary
+    """
+    def __init__(self):
+        self.g = {}
+
+    def nodes(self):
+        """
+        returns a list of nodes
+        """
+        return self.g.keys()
+
+    def edges(self):
+        """
+        returns a list of tuples the corresponds to the edges
+        """
+        result = []
+        for key, values in self.g.iteritems():
+            for value in values:
+                if value:
+                    result.append((key, value))
+        return result
+
+    def add_node(self, n):
+        """
+        Add a node to the Graph
+        """
+        self.g[n] = []
+
+    def add_edge(self, n1, n2):
+        """
+        Add an edge between two nodes
+
+        If any of the nodes were not created then they will be created
+        """
+        if not self.g.has_key(n1):
+            self.add_node(n1)
+        if not self.g.has_key(n2):
+            self.add_node(n2)
+        self.g[n1].append(n2)
+
+    def del_node(self, n):
+        """
+        Delete an existing node from the Graph
+        """
+        try:
+            del self.g[n]
+            for node in self.g.iterkeys():
+                if n in self.g[node]:
+                    self.g[node].remove(n)
+        except KeyError:
+            raise IndexError("Node doesn't exist")
+
+    def del_edge(self, n1, n2):
+        """
+        Delete an edge from the Graph. If an edge doesn't exist pass
+        """
+        try:
+            self.g[n1].remove(n2)
+        except ValueError:
+            pass
+        except KeyError:
+            raise IndexError("First node doesn't exist")
+
+    def has_node(self, n):
+        """
+        Return if Graph has a node
+        """
+        return self.g.has_key(n)
+
+    def neighbors(self, n):
+        """
+        Return the edges for a node in a list
+        """
+        try:
+            return self.g[n]
+        except KeyError:
+            raise IndexError("Node doesn't exist")
+
+    def adjacent(self, n1, n2):
+        """
+        Return if an edge exist between two nodes
+        """
+        try:
+            return n2 in self.g[n1]
+        except KeyError:
+            raise IndexError("One or more nodes doesn't exist")
+
+    def depth_first_traversal(self, node,result=""):
+        if not str(node) in result:
+            result = '{}{}'.format(result, node)
+            if self.g[node]:
+                for edge_node in self.g[node]:
+                    result = self.depth_first_traversal(edge_node,result)
+        return result
```

Comment:
Your result here is a string, which is nice and presentable, but isn't good for much except looking at.  If you returned a value like a list, then you could print it, but also use it for other purposes.  You'll want to think about that.  In addition, if you've got unicode values in your graph, and any of them contain characters outside the ASCII plain, you're in for a `UnicodeEncodeError`